### ISSUE TITLE
Can probably eliminate sort in RealignIndels

### ISSUE THREAD

**Issue by fnothaft** [bigdatagenomics/adam#1137]
We have a sort in the INDEL realigner that takes most of the realignment time. We do this so that we can get the full alignment position of all reads that cover a target, but this is actually an unnecessary step, since we join the reads back later and just check for overlap. If we eliminate this, we should improve INDEL realignment runtime by ~60% with negligible impact on accuracy.

I'm thinking that I'll support this via a flag.

**Comment by NeillGibson** [comment]
GATK and Freebayes handle indel realignment inside the variant caller, eliminating the extra realign indels step. 

FreeBayes documentation:
```
Indel realignment is accomplished internally using a read-independent method, 
and issues resulting from discordant alignments are dramatically reducedy through the direct detection of haplotypes.
```

GATK documentation:
```
Note that indel realignment is no longer necessary for variant discovery if you plan to use a variant caller that performs a haplotype assembly step, 
such as HaplotypeCaller or MuTect2. 
However it is still required when using legacy callers such as UnifiedGenotyper or the original MuTect.
```

If the variant caller(s) that you plan to use with Adam also are haplotype aware this would mean that you could drop the indel realignment step/tool.

**Comment by iskandr** [comment]
@fnothaft Do you think your indel realigner would work with spliced RNA reads? (GATK drops those reads)

**Comment by iskandr** [comment]
@ihodes ^ maybe another option for us? I'll try to take a subset of one of our B6.F10 (mouse melanoma) BAMs as example input.

**Comment by fnothaft** [comment]
That'd be great! I can take a look.

**Comment by fnothaft** [comment]
This will be resolved after #1324.

**Comment by fnothaft** [comment]
Resolved by #1324.
